### PR TITLE
Extract observer to an independent docker image

### PR DESCRIPTION
**Pull request by laodouya**
1. observer image is now base on covenantsql/explorer image
2. refactor build.sh, move to makefile

Now use `make` instead of ~~`build.sh`~~. And use `make use_all_cores` for building in full cpu cores.

### CODE REVIEW

File: docker/observer.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Stage: observer
+FROM covenantsql/explorer
+
+ARG COMMIT
+ARG VERSION
+
+LABEL \
+    name="covenantsql_observer" \
```
Comment (auxten) [inline docker/observer.Dockerfile:8]: Use the `covenantsql/explorer` name

File: docker-compose.yml [hunk 2]
```diff
@@ -175,20 +151,30 @@ services:
...
+    container_name: covenantsql_observer
     restart: always
     ports:
+      - "11106:4663"
       - "11108:80"
+    command:
+      [
+        "-listen",
+        "0.0.0.0:4663",
+      ]
```
Comment (xq262144) [inline docker-compose.yml]: Should make this command be the default Dockerfile command.

File: docker-compose.yml [hunk 3]
```diff
@@ -175,20 +151,30 @@ services:
...
-    image: covenantsql/explorer:latest
-    container_name: covenantsql_explorer
-    depends_on:
-      - covenantsql_observer
+  covenantsql_observer:
+    image: covenantsql/covenantsql-observer:latest
+    container_name: covenantsql_observer
     restart: always
     ports:
+      - "11106:4663"
```
Comment (xq262144) [inline docker-compose.yml]: Forwarding port 4663 is no longer required.

File: docker-compose.yml [hunk 4]
```diff
@@ -175,20 +151,30 @@ services:
...
       - "11108:80"
+    command:
+      [
+        "-listen",
+        "0.0.0.0:4663",
+      ]
     environment:
-      COVENANTSQL_EXPLORER_DOMAIN: localhost
-      COVENANTSQL_OBSERVER_ADDR: covenantsql_observer:4663
```
Comment (xq262144) [inline docker-compose.yml:188]: Should override `COVENANTSQL_OBSERVER_ADDR` to `localhost:4663`

File: bin/docker-entry.sh [hunk 5]
```diff
@@ -12,7 +12,9 @@ blockproducer)
     ;;
 observer)
     rm -f /app/node_observer/observer.db
-    exec /app/cql-observer -config "${COVENANT_CONF}" "${@}"
+    MAGIC_DOLLAR='$' envsubst < /etc/nginx/conf.d/servers/explorer.conf.template > /etc/nginx/conf.d/default.conf
+    nohup nginx -g 'daemon off;' 1>access.log 2>error.log &
```
Comment (xq262144) [inline bin/docker-entry.sh]: Do not store log files in container, it would sooner or later consume whole disk space.

**Review by xq262144 — APPROVED** [review]
Approve

**Review by leventeliu — APPROVED** [review]
LGTM.

**Pull request merged**